processing/src/main/java/io/druid/segment/filter/Filters.java

```diff
@@ -149,4 +155,171 @@ public void remove()
         }
     );
   }
+
+  public static Filter convertToCNFFromQueryContext(Query query, Filter filter)
+  {
+    if (filter == null) {
+      return null;
+    }
+    boolean useCNF = query.getContextBoolean(CTX_KEY_USE_FILTER_CNF, false);
+    return useCNF ? convertToCNF(filter) : filter;
+  }
+
+  public static Filter convertToCNF(Filter current)
+  {
+    current = pushDownNot(current);
+    current = flatten(current);
+    current = convertToCNFInternal(current);
+    current = flatten(current);
+    return current;
+  }
+
+  // copied from apache hive
```

Comment:
added section to NOTICE, added comment in this file pointing to the Hive original source